content-src/asrouter/templates/CFR/templates/ExtensionDoorhanger.schema.json

```diff
@@ -0,0 +1,131 @@
+{
+  "title": "ExtensionDoorhanger",
+  "description": "A template with a heading, addon icon, title and description. No markup allowed.",
+  "version": "1.0.0",
+  "type": "object",
+  "definitions": {
+    "plainText": {
+      "description": "Plain text (no HTML allowed)",
+      "type": "string"
+    },
+    "linkUrl": {
+      "description": "Target for links or buttons",
+      "type": "string",
+      "format": "uri"
+    }
+  },
+  "properties": {
+    "notification_text": {
+      "allOf": [
+        {"$ref": "#/definitions/plainText"},
+        {"description": "Text for location bar chiclet."}
+      ]
+    },
+    "info_icon": {
+      "type": "object",
+      "properties": {
+        "label": {
+          "allOf": [
+            {"$ref": "#/definitions/plainText"},
+            {"description": "Text for button tooltip used to provider information about the doorhanger."}
+          ]
+        },
+        "url": {
+          "allOf": [
+            {"$ref": "#/definitions/linkUrl"},
+            {"description": "Link to support page with the information about the doorhanger."}
+          ]
+        }  
+      }
+    },
+    "heading_text": {
+      "allOf": [
+        {"$ref": "#/definitions/plainText"},
+        {"description": "Doorhanger heading describing its purpose."}
+      ]
+    },
+    "addon": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "allOf": [
+            {"$ref": "#/definitions/plainText"},
+            {"description": "Addon name"}
+          ]
+        },
+        "author": {
+          "allOf": [
+            {"$ref": "#/definitions/plainText"},
+            {"description": "Addon author"}
+          ]
+        },
+        "icon": {
+          "allOf": [
+            {"$ref": "#/definitions/linkUrl"},
+            {"description": "Addon icon"}
+          ]
+        }
+      }
+    },
+    "text": {
+      "allOf": [
+        {"$ref": "#/definitions/plainText"},
+        {"description": "Description for the addon."}
+      ]
+    },
+    "amo_url": {
+      "allOf": [
+        {"$ref": "#/definitions/linkUrl"},
+        {"description": "Link that offers more information related to the addon."}
+      ]
+    },
+    "buttons": {
+      "type": "object",
+      "properties": {
+        "primary": {
+          "type": "object",
+          "properties": {
+            "label": {
+              "allOf": [
+                {"$ref": "#/definitions/plainText"},
+                {"description": "Text for the primary button of the doorhanger."}
+              ]
+            },
+            "action": {
+              "type": "object",
+              "properties": {
+                "type": {
+                  "enum": ["INSTALL_ADDON_FROM_URL"],
```

Comment:
technically this would be an `enum` of every action but maybe just `string` for now?